File: prime-factors/python/prime_factors.py

class PrimeFactors:
    @staticmethod
    def generate(n):
        primes = []
        candidate = 2
        while (n > 1):
            while (n % candidate == 0):
                primes.append(candidate)
                n = n / candidate
            candidate = candidate + 1
        return primes

File: prime-factors/java/primefactors/src/main/java/com/github/reeda/kata/PrimeFactors.java
package com.github.reeda.kata;

import com.google.common.collect.Lists;
import java.util.List;

public class PrimeFactors {
    public static List<Integer> generate(int n) {
        List<Integer> factors = Lists.newArrayList();
        for (int candidate = 2; n > 1; candidate++) {
            for (; n % candidate == 0; n /= candidate) {
                factors.add(candidate);
            }
        }
	    return factors;
    }
}
